cmd/cogito/main_test.go

```diff
@@ -1,13 +1,113 @@
 package main
 
 import (
+	"bytes"
+	"io"
+	"strings"
 	"testing"
 
 	"gotest.tools/v3/assert"
 )
 
-func TestRunSmoke(t *testing.T) {
-	err := run(nil, nil, []string{"foo"})
+func TestRunSmokeSuccess(t *testing.T) {
+	type testCase struct {
+		name    string
+		args    []string
+		in      string
+		wantOut string
+	}
 
+	test := func(t *testing.T, tc testCase) {
```

Comment:
This is a new way to write table-driven tests that I got from https://github.com/gotestyourself/gotest.tools/wiki/Go-Testing-Patterns#table-driven-tests.

The idea is to first show the shape of the test input (struct testCase line 13) and just after (line 20 here) _what_ we are testing.

Only after the test function we enumerate the test cases (start at line 31).